RationaleMCP/0031/concept.md

```diff
@@ -0,0 +1,135 @@
+   1. an optional description string and
+   1. an optional annotation.
+* The dimension of an array is defined by a list of dimension sizes inside square brackets, e.g. `[2,3]`. The dimension sizes are allowed to be: 
+   * integer literals,
+   * `:`. In this case is determined from the dimension of the right side of the declaration equation, if exist and if it is possible. If not, the size is seen as unknown and no subscription is possible.
+   * Only in functions: expressions containing literals and references to input parameters and other (predefined) pure functions
+* Parameters and Constants from Modelica are mapped to variables with special annotations.
+* It is assumed, that constants and parameters of the Modelica model was evaluated already evaluated if needed, i.e. because dimensions or branches or loops in equations depend from them.
+* fixed start attributes are converted to initial equations, non fixed start attributes to annotations
+
+### Expressions
+
+* Expressions are defined as in Modelica.
+* There are the predefined operators and functions of Modelica and some more to support runtime behavior, e.g. event handling
+* There is no implied behavior of relations and builtin operators with regard to event handling any more. All has to be made explicit by introduction of helper variables and equations.
+* The operators `pre(...)`, `previous(...)`, `sample(...)`, `delay(...)`, `spatialDistribution(...)` have the same semantics as in Modelica.
+* Function calls have no default arguments, all must be explicitly specified in the correct order. There are no named arguments.
+
+### Equations and Algorithms 
+* Equation and algorithm sections are defined as in Modelica, including initial equations and algorithms.
+* extension: equation and algorithm sections may have a dimension (to retain the information about vectorized components).
+* There is no special initialization semantics of algorithms as in Modelica (this has to be made explicit by assignments)
+* there are no `when` clauses in equaitons and algorithms. They must be transformed in if/then/else clauses.
+
+### Functions
+
+* Functions are defined as in Modelica.
+* They contain exactly one algorithm section.
+* Partial functions are not allowed.
+* Input argument variables do not have declaration equations. 
+* Functions may have an external specification, referring to some external resources (e.g. shared libraries, files etc.).
+* If a function has functional inputs, for each different call a new function type without them is generated. _(this can be relaxed later if functional types are introduced)_
+
+### Annotations
+
+* Annotations are defined as in Modelica.
+* They are `<name> = <expression>` pairs with hierarchical structure,
+* An annotation can have sub annotations, which are a recursive list of annotations inside parenthesis, separated by colons.
+* There are many more defined annotation names than in Modelica to hold additional information.
+
+### Model Description
```

Comment:
Can we discuss this in the context of https://github.com/modelica/ModelicaSpecification/blob/MCP/0031/RationaleMCP/0031/grammar.md instead of here?

Reply:
Shure.